src/store/notification/index.ts

```diff
@@ -0,0 +1,87 @@
+import Vue from 'vue'
+import {ActionTree, Module} from 'vuex'
+import {IRootState} from '@/store'
+import {ToastOptions} from 'vue-toastification/dist/types/src/types'
+import {POSITION} from 'vue-toastification/src/ts/constants'
+import {TYPE} from 'vue-toastification/dist/types/src/ts/constants'
+
+/**
+ * For more details about available options, see https://vue-toastification.maronato.dev/
+ */
+const GENERAL_OPTIONS: ToastOptions = {
+  position: POSITION.TOP_RIGHT,
+  draggable: true,
+  draggablePercent: 0.6,
+  closeOnClick: false,
+  timeout: false,
+}
+
+/**
+ * Note, we can use a custom component for `content`, not only texts
+ */
+export type IToastContent = string | object
+
+export interface IToast {
+  content: IToastContent,
+  options: ToastOptions,
+}
+
+export interface INotificationState {}
+
+export const stateFactory = (): INotificationState => ({})
+
+export const notificationActions: ActionTree<INotificationState, IRootState> = {
+  toast(_context, {content, options}: IToast) {
+    Vue.$toast(content, {
+      ...GENERAL_OPTIONS,
+      ...options,
+    } as ToastOptions)
+  },
+  toastError({dispatch}, {content, options}: IToast) {
+    const defaultOptions: ToastOptions = {} as ToastOptions
+    Vue.$toast.error(content, {
+      ...GENERAL_OPTIONS,
+      ...defaultOptions,
+      ...options,
+    } as (ToastOptions & {
+      type?: TYPE.ERROR | undefined
+    }) | undefined)
```

Comment:
about the `as (ToastOptions & {type?: TYPE.ERROR | undefined}) | undefined`

I think you can also do something like this:
```
import {TYPE as ToastType} from 'vue-toastification/dist/types/src/ts/constants'

export interface IToast<T extends ToastType> {
  content: IToastContent,
  options: ToastOptions & {type: T},
}

toast(_context, {content, options}: IToast<ToastType>) {
    Vue.$toast(content, {
      ...GENERAL_OPTIONS,
      ...options,
    })
  },
  toastError({dispatch}, {content, options}: IToast<ToastType.ERROR>) {
    Vue.$toast.error(content, {
      ...GENERAL_OPTIONS,
      ...options,
    })
  },
```

Reply:
But this makes the `type` required. Let me think a bit more 🙂